[BUG] Validation error using activity that consumes an item use.

**Description**
A clear and concise description of what the bug is, along with any other context or screenshots about the problem.

**Steps to Reproduce**
1. Import [fvtt-Item-rsr-bug-item-8RjpQDylJ3z9USNp.json](https://github.com/user-attachments/files/17486752/fvtt-Item-rsr-bug-item-8RjpQDylJ3z9USNp.json)
2. Add item to the inventory of a token.
3. Activate the "Attack" activity.
4. Ensure "Consume Item Use?" checkbox is checked
5. Click Use Ability.
6. A validation error message is shown and logged to the console:
```
foundry.js:72076 validation errors:
  system: 
    quantity: must be an integer
```
```
foundry.js:655 Error: undefined. Item5e [bXw06ibih4DzMRxd] validation errors:
  system: 
    quantity: must be an integer
[Detected 1 package: system:dnd5e(4.0.4)]
    at DataModelValidationFailure.asError (foundry-esm.js:4710:14)
    at Item5e.validate (foundry-esm.js:10432:61)
    at Item5e.updateSource (foundry-esm.js:10502:12)
    at #preUpdateDocumentArray (foundry-esm.js:56521:22)
    at async ClientDatabaseBackend._updateDocuments (foundry-esm.js:56469:7)
    at async Item5e.updateDocuments (foundry-esm.js:11473:23)
    at async #applyUsageUpdates (mixin.mjs:457:49)
    at async AttackActivity.consume (mixin.mjs:353:22)
    at async AttackActivity.use (mixin.mjs:241:21)
```

**Versions**
 - Module: [3.3.2]
 - dnd5e: [4.0.4]
 - Foundry: [12.331]

**Other Modules**
Happens only with RSR enabled and doesn't occur without it.

If "Consume Item Use?" is unchecked, error doesn't occur.

--- a/src/utils/hooks.js
+++ b/src/utils/hooks.js
@@ -141,8 +141,10 @@ export class HooksUtility {
 
             Hooks.on(HOOKS_DND5E.ACTIVITY_CONSUMPTION, (activity, usageConfig, messageConfig, updates) => {
                 if (activity.hasOwnProperty(ROLL_TYPE.ATTACK) && updates.item.length > 0 && messageConfig.data) {
-                    messageConfig.data.flags[MODULE_SHORT].ammunition = updates.item[0]._id;
-                    updates.item[0]["system.quantity"]++;
+                    const ammo = updates.item.find(i => i["system.quantity"]);
+                    if (!ammo) return;
+                    messageConfig.data.flags[MODULE_SHORT].ammunition = ammo._id;
+                    ammo["system.quantity"]++;
                 }
             });
         }

